### PR TITLE
Api detail book

### PR DESCRIPTION
**Pull request by at-minhdao**
## Description
- API get detail of book
## Reference Requirements
- Product Backlog ID: PBI-38
- Task ID: #38.2

## Migrations
NO

## Impacted Areas in Application
List general components of the application that this PR will affect:
- None

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)
```mysql
select `books`.`id`, `books`.`title`, `books`.`category_id`, `books`.`description`, 
           `books`.`language`, `books`.`rating`, `books`.`total_rating`, `books`.`picture`, 
           `books`.`author`, `books`.`price`, `books`.`unit`, `books`.`year`, `books`.`page_number`, 
           `borrows`.`status`, `users`.`id` as `user_id`, `users`.`name` as `donator` from `books` 
left join `borrows` on `books`.`id` = `borrows`.`book_id` 
inner join `users` on `books`.`from_person` = `users`.`employ_code` 
where `books`.`id` = '171' and `books`.`deleted_at` is null 
order by `borrows`.`created_at` desc limit 1
```
```
*************************** 1. row ***************************
           id: 1
  select_type: SIMPLE
        table: books
   partitions: NULL
         type: const
possible_keys: PRIMARY,books_from_person_foreign
          key: PRIMARY
      key_len: 4
          ref: const
         rows: 1
     filtered: 100.00
        Extra: Using temporary; Using filesort
*************************** 2. row ***************************
           id: 1
  select_type: SIMPLE
        table: borrows
   partitions: NULL
         type: ref
possible_keys: borrows_book_id_foreign
          key: borrows_book_id_foreign
      key_len: 4
          ref: const
         rows: 2
     filtered: 100.00
        Extra: NULL
*************************** 3. row ***************************
           id: 1
  select_type: SIMPLE
        table: users
   partitions: NULL
         type: ref
possible_keys: employ_code
          key: employ_code
      key_len: 42
          ref: const
         rows: 1
     filtered: 100.00
        Extra: NULL
```
- Refer to [Using Explain with MySQL](https://viblo.asia/p/su-dung-explain-de-toi-uu-cau-lenh-mysql-BYjv44gmvxpV)

### CODE REVIEW

**Comment by at-hieule** [comment]
Description  trong PR làm gì vậy?

File: routes/api.php [hunk 1]
```diff
@@ -16,3 +16,5 @@
 Route::middleware('auth:api')->get('/user', function (Request $request) {
     return $request->user();
 });
+
+Route::get('books/{book}', 'Api\BookController@show');
```
Comment (at-hieule) [inline routes/api.php]: Nên cho `api` vào `namespace` lun hì

**Review by at-hieule — COMMENTED** [review]
please check my cmt

**Comment by at-hieule** [comment]
OK

File: app/Exceptions/Handler.php [hunk 2]
```diff
@@ -50,6 +52,19 @@ public function report(Exception $exception)
      */
     public function render($request, Exception $exception)
     {
+        if ($request->route()->getPrefix() === 'api') {
+            if ($exception instanceof ModelNotFoundException) {
+                return response()->json([
+                    'meta' => [
+                        'status' => 'failed',
+                        'code' => 404,
```
Comment (at-duyho) [inline app/Exceptions/Handler.php]: - you should use Response::HttpNotfound

File: app/Exceptions/Handler.php [hunk 3]
```diff
@@ -50,6 +52,19 @@ public function report(Exception $exception)
      */
     public function render($request, Exception $exception)
     {
+        if ($request->route()->getPrefix() === 'api') {
+            if ($exception instanceof ModelNotFoundException) {
+                return response()->json([
+                    'meta' => [
+                        'status' => 'failed',
```
Comment (at-duyho) [inline app/Exceptions/Handler.php]: - message should define in language file

File: app/Exceptions/Handler.php [hunk 4]
```diff
@@ -50,6 +52,19 @@ public function report(Exception $exception)
...
+                        'code' => 404,
+                    ],
+                    'error' => [
+                        'message' => 'Page not found!',
```
Comment (at-duyho) [inline app/Exceptions/Handler.php]: message should define in language file

**Review by ghost — CHANGES_REQUESTED** [review]
- check my comment.
- check query explain. You miss some query and why explain type is null?

File: app/Http/Controllers/Api/BookController.php [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+                    ->with(['category' => function ($query) {
+                        $query->select('id', 'title');
+                    }])
+                    ->leftJoin('borrows', 'books.id', '=', 'borrows.book_id')
+                    ->join('users', 'books.from_person', '=', 'users.employ_code')
+                    ->orderBy('borrows.created_at', 'DESC')
+                    ->findOrFail($id);
+                    
+        if ($book) {
+            $book->picture = url('/').'/'.config('define.books.folder_store_books').$book->picture;
```
Comment (at-duyho) [inline app/Http/Controllers/Api/BookController.php]: - nên dùng setPictureAttribute function trong model.

**Review by ghost — CHANGES_REQUESTED** [review]
- check my comment

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**